src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -284,206 +425,247 @@ private void hoeIfAble(final BlockPos position, final Field field)
             world.setBlockState(position, Blocks.FARMLAND.getDefaultState());
             worker.damageItemInHand(1);
             mineBlock(position.up());
+            return true;
         }
+        return false;
     }
 
     /**
-     * Checks if the ground should be planted.
+     * Checks if the ground should be hoed and the block above removed.
      *
      * @param position the position to check.
      * @param field    the field close to this position.
      * @return true if should be hoed.
      */
-    private boolean shouldPlant(@NotNull final BlockPos position, @NotNull final Field field)
-    {
-        @Nullable final ItemStack itemStack = BlockUtils.getItemStackFromBlockState(world.getBlockState(position.up()));
-
-        if (itemStack != null && itemStack.getItem() == field.getSeed())
-        {
-            requestSeeds = false;
-        }
-
-        return !field.isNoPartOfField(world, position) && !(world.getBlockState(position.up()).getBlock() instanceof BlockCrops)
-                 && !(world.getBlockState(position).getBlock() instanceof BlockHutField) && world.getBlockState(position).getBlock() == Blocks.FARMLAND;
-    }
-
-    /**
-     * Plants the crop at a given location.
-     *
-     * @param item     the crop.
-     * @param position the location.
-     */
-    private boolean plantCrop(Item item, @NotNull BlockPos position)
+    private boolean shouldHoe(@NotNull final BlockPos position, @NotNull final Field field)
     {
-        final int slot = worker.findFirstSlotInInventoryWith(item);
-        if (slot == -1)
-        {
-            return false;
-        }
-        else
-        {
-            @NotNull final IPlantable seed = (IPlantable) item;
-            world.setBlockState(position.up(), seed.getPlant(world, position));
-            getInventory().decrStackSize(slot, 1);
-            requestSeeds = false;
-            //Flag 1+2 is needed for updates
-            return true;
-        }
+        return !field.isNoPartOfField(world, position)
+                 && !BlockUtils.isBlockSeed(world, position.up())
+                 && !(world.getBlockState(position).getBlock() instanceof BlockHutField)
+                 && (world.getBlockState(position).getBlock() == Blocks.DIRT || world.getBlockState(position).getBlock() == Blocks.GRASS);
     }
 
     /**
-     * Resets the basic variables of the class.
+     * Sets the hoe as held item.
      */
-    private void resetVariables()
+    private void equipHoe()
     {
-        requestSeeds = true;
-        shouldDumpInventory = true;
-        shouldTryToGetSeed = true;
+        worker.setHeldItem(getHoeSlot());
     }
+    
+    private int dist = 0;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1213)